### PR TITLE
Add support for git dependencies

### PR DESCRIPTION
**Pull request by muyajil**
### Description

- Add support for `git+` prefixed pip dependencies

### CODE REVIEW

File: conda_lock/src_parser/pyproject_toml.py [hunk 1]
```diff
@@ -112,6 +114,57 @@ def poetry_version_to_conda_version(version_string: Optional[str]) -> Optional[s
...
+def handle_mapping(
+    depattrs: collections.abc.Mapping,
+    depname: str,
+    path: pathlib.Path,
+    category: str,
+    in_extra: bool,
+    default_category: str,
+    manager: Literal["conda", "pip"],
+    poetry_version_spec: Optional[str],
+) -> Tuple[Optional[str], Literal["conda", "pip"], list, Optional[str]]:
```
[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply conda_lock/src_parser/pyproject_toml.py:126]: Additionally a NamedTuple or a pydantic model is probably a better return type here.

**Review by mariusvniekerk — COMMENTED** [review]
Looks very nice, just a few minor comments

**Comment by muyajil** [comment]
@mariusvniekerk  thx for the notes! I pushed an update

**Comment by muyajil** [comment]
We have some workflows depending on this. If I can do sth to help get this merged let me know :) Otherwise I would appreciate it a lot if we can move forward with this.

**Pull request merged**